src/monitor.rs

```diff
@@ -334,31 +334,31 @@ where
 	Ok(())
 }
 
-async fn ensure_no_better_solution(
+async fn ensure_solution_passes_strategy(
 	api: &SubxtClient,
 	at: Hash,
 	score: sp_npos_elections::ElectionScore,
 	strategy: SubmissionStrategy,
 ) -> Result<(), Error> {
-	let epsilon = match strategy {
-		// don't care about current scores.
-		SubmissionStrategy::Always => return Ok(()),
-		SubmissionStrategy::IfLeading => Perbill::zero(),
-		SubmissionStrategy::ClaimBetterThan(epsilon) => epsilon,
-	};
+	// don't care about current scores.
+	if matches!(strategy, SubmissionStrategy::Always) {
+		return Ok(())
+	}
 
 	let addr = runtime::storage().election_provider_multi_phase().signed_submission_indices();
 	let indices = api.storage().fetch_or_default(&addr, Some(at)).await?;
 
 	log::debug!(target: LOG_TARGET, "submitted solutions: {:?}", indices.0);
 
-	for (other_score, _) in indices.0 {
-		if !score.strict_threshold_better(other_score, epsilon) {
-			return Err(Error::BetterScoreExist)
-		}
+	if indices
```

Comment:
//cc @kianenigma you added [this check](https://github.com/paritytech/polkadot/blob/6c3d5add1c53f3daff6800420c7e65a18a0f0ed1/utils/staking-miner/src/monitor.rs#L123-#L126)

I missed it in the review but why is that needed? AFAIU, the solution is better than solutions is in queue according the "strategy" and if that's ttrue then the solution with worse will be kicked out?

Reply:
yeah I checked the original PR as well and it doesn't make sense, the whole `if` has to go away. Can you make sure it is fixed in Polkadot as well?

Reply:
sure